python/kserve/kserve/model.py

```diff
@@ -323,6 +340,10 @@ async def _http_predict(
             data=payload,
             headers=predict_headers,
         )
+
+        if response_header is not None:
+            response_header.update(response.headers)
```

Comment:
We might need a flag to determine if kserve should pass down the response headers from predictor.